### PR TITLE
Add adviser component

### PR DESCRIPTION
**Pull request by plisovin**
### Trello card

[Trello-4696](https://trello.com/c/XpL1Ymc3/4696-embed-adviser-form-first-step-on-tta-and-eta-pages)

### Context

Now that the adviser funnel has been moved to GIT, we can embed the first step on appropriate pages around the site to encourage people to start entering data into the funnel.

We want to do this as we know from doing this with the mailing list on the home page and landing pages that doing so increases conversion rate.

### Changes proposed in this pull request

A new component has been added to allow to use it on other pages

### CODE REVIEW

**Comment by plisovin** [comment]
> @EllieNodder has there been any discussion to add any content to support embedding the adviser sign-up funnels into the teacher training and explore teaching adviser pages? It might be just me, but this feels a little lost at the end of the page?

It's still work in progress. I thought I just put it there as placeholder to prove this component works

**Comment by EllieNodder** [comment]
> > @EllieNodder has there been any discussion to add any content to support embedding the adviser sign-up funnels into the teacher training and explore teaching adviser pages? It might be just me, but this feels a little lost at the end of the page?
> 
> It's still work in progress. I thought I just put it there as placeholder to prove this component works

cool - let me know when you want any content input :)

**Comment by plisovin** [comment]
> https://review-get-into-teaching-app-3472.london.cloudapps.digital

> > 
> > 
> > It's still work in progress. I thought I just put it there as placeholder to prove this component works
> 
> cool - let me know when you want any content input :)

This page is ready https://review-get-into-teaching-app-3472.london.cloudapps.digital/teacher-training-advisers with the first step form at the bottom. Can you suggest where do we want this form move to? 

And the second page https://review-get-into-teaching-app-3472.london.cloudapps.digital/explore-teaching-advisers wil lbe similar

**Comment by EllieNodder** [comment]
> > https://review-get-into-teaching-app-3472.london.cloudapps.digital
> 
> > > It's still work in progress. I thought I just put it there as placeholder to prove this component works
> > 
> > 
> > cool - let me know when you want any content input :)
> 
> This page is ready https://review-get-into-teaching-app-3472.london.cloudapps.digital/teacher-training-advisers with the first step form at the bottom. Can you suggest where do we want this form move to?
> 
> And the second page https://review-get-into-teaching-app-3472.london.cloudapps.digital/explore-teaching-advisers will be similar

Could the form go on /teacher-training-advisers between the text 'If you got your qualifications outside the UK, call us on 0800 389 2500 to check if you're eligible for an adviser.' and 'If you're a veteran'?

And on /explore-teaching-advisers, could it go where the green 'Get an explore teaching adviser' button is?

I'm also wondering if it's possible for the component to look a bit more like it does on the /landing/how-much-do-teachers-get-paid page, like this?

image

I think the purple h2 gives users a bit more context about what they're signing up for (and I'm wondering if we might need a heading above the form for accessibility reasons?)

**Comment by plisovin** [comment]
> And on /explore-teaching-advisers, could it go where the green 'Get an explore teaching adviser' button is?
> 
> I'm also wondering if it's possible for the component to look a bit more like it does on the /landing/how-much-do-teachers-get-paid page, like this?

> 
> I think the purple h2 gives users a bit more context about what they're signing up for (and I'm wondering if we might need a heading above the form for accessibility reasons?) @plisovin

@EllieNodder can you please check the updated version of this page?

**Comment by EllieNodder** [comment]
> > And on /explore-teaching-advisers, could it go where the green 'Get an explore teaching adviser' button is?
> > I'm also wondering if it's possible for the component to look a bit more like it does on the /landing/how-much-do-teachers-get-paid page, like this?
> 
> > I think the purple h2 gives users a bit more context about what they're signing up for (and I'm wondering if we might need a heading above the form for accessibility reasons?) @plisovin
> 
> @EllieNodder can you please check the updated version of this page?

positioning looks good to me! is there any way we could reduce the amount of white space between the sign up form and the text above?

Also, the copy in the sign up form will need tweaking - I leave in the capable hands of @philspray when he returns

**Comment by plisovin** [comment]
> 
> positioning looks good to me! is there any way we could reduce the amount of white space between the sign up form and the text above?
> 
> Also, the copy in the sign up form will need tweaking - I leave in the capable hands of @philspray when he returns

I've fixed the space around the form

**Comment by gemmadallmandfe** [comment]
@plisovin I've tested the component on the TTA page and the ETA page and checked with Tejas that the data appears in CRM and on the adviser dashboards - all looks good

Just needs some revised content from @philspray when he is back next week

File: app/views/content/explore-teaching-advisers/_ending.html.erb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<div class="row inset">
```
Comment (gemmadallmandfe) [inline app/views/content/explore-teaching-advisers/_ending.html.erb:1]: This section doesn't appear on the front end @plisovin - it should?

https://review-get-into-teaching-app-3472.london.cloudapps.digital/explore-teaching-advisers

Instead the teacher-training-advisers ending appears instead (sections titled `If you are a veteran` and `Not a final year student or graduate`? This needs removing and the explore-teaching-advisers ending needs adding.

**Review by gemmadallmandfe — COMMENTED** [review]
@plisovin I've found a few issues with the TTA and ETA pages - could you have a look this week so they are in a good place for when Phil is back next week? Thanks!

File: app/views/content/teacher-training-advisers/_ending.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<div class="row inset">
```
[the remark this answers is not in the document]
Comment (plisovin) [reply app/views/content/teacher-training-advisers/_ending.html.erb:1]: @gemmadallmandfe do you have an example of a page? I copied this from the similar page

File: app/views/content/teacher-training-advisers/_ending.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  <section class="col col-720">
+    <h3>If you're a veteran<h3>
+    <p>If you're a veteran, you do not need to have or be studying for a degree to get support from an adviser.</p>
+    <p><a href="tel:08003892500">Call us on 0800 389 2500</a> to get an adviser.</p>
+    <p>Our telephone service is available Monday to Friday between 8:30am and 5:30pm, except on [bank holidays](https://www.gov.uk/bank-holidays).</p>
```
[the remark this answers is not in the document]
Comment (plisovin) [reply app/views/content/teacher-training-advisers/_ending.html.erb]: Fixed

File: app/views/content/explore-teaching-advisers/_ending.html.erb (hunk 1 above)
Comment (plisovin) [reply app/views/content/explore-teaching-advisers/_ending.html.erb:1]: > Instead the teacher-training-advisers ending appears instead (sections titled `If you are a veteran` and `Not a final year student or graduate`? This needs removing and the explore-teaching-advisers ending needs adding.

Sorry, was wrong partial used in ETA, fixed now

**Comment by gemmadallmandfe** [comment]
@plisovin Thanks for fixing the page endings / h3 / links - all retested successfully :-)

Couple more things - I noticed the padding on TTA and ETA pages is a bit different to these new pages, plus the TTA page is now more central on desktop

Paging @garethwilliamsdesign for advice on whether these new versions need tweaking and if so, how:

**ETA page**
* https://getintoteaching.education.gov.uk/explore-teaching-advisers
* https://review-get-into-teaching-app-3472.london.cloudapps.digital/explore-teaching-advisers

For example:
image
image

**TTA page**
* https://getintoteaching.education.gov.uk/teacher-training-advisers
* https://review-get-into-teaching-app-3472.london.cloudapps.digital/teacher-training-advisers

For example:
image
image

**Comment by plisovin** [comment]
> @plisovin Thanks for fixing the page endings / h3 / links - all retested successfully :-)

Pushed some changes

File: app/views/content/teacher-training-advisers.md [hunk 4]
```diff
@@ -1,129 +1,13 @@
...
-  get-a-tta:
-    name: simple
-    arguments:
-      title: Get an adviser
-      text: "Answer a few questions about your situation and we'll match you with an adviser."
-      link_text: "Get a teacher training adviser"
-      link_target: "/teacher-training-adviser/sign_up/identity"
-      icon: "icon-person"
+talk_to_us: true
+noindex: true
```
Comment (gemmadallmandfe) [inline app/views/content/teacher-training-advisers.md:12]: @plisovin We don't want to noindex the TTA page - can you change this please?

File: app/views/content/teacher-training-advisers/_article.html.erb [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+    <ul>
+      <li>brush up on your subject knowledge</li>
+      <li>find a role</li>
+    </ul>
+
+      <%= render Content::QuoteListComponent.new(
+        quotes: [{
+          heading: "Stephen, returning teacher",
+          text: "[My teacher training adviser] took considerable time to understand my situation..",
```
Comment (gemmadallmandfe) [inline app/views/content/teacher-training-advisers/_article.html.erb:98]:
```suggestion
          text: "[My teacher training adviser] took considerable time to understand my situation.",
```
@philspray

File: app/views/content/teacher-training-advisers/_article.html.erb [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      <%= render Content::QuoteListComponent.new(
+        quotes: [{
+          heading: "Habiba, teacher with a hearing impairment",
+          text: "The application process went smoothly thanks to the help of my teacher training adviser, Ellie.",
+          background: "grey",
+          accreditation: "<a href=\"/blog/overcoming-challenges-to-become-a-teacher-hearing-impairment\">Read how Habiba successfully applied to train to teach</a>",
+        }]
+      ) %>
+
+    <h3>If you've applied and been unsuccessful</h3>
```
Comment (gemmadallmandfe) [inline app/views/content/teacher-training-advisers/_article.html.erb:65]: @plisovin @philspray 

The apostrophe on this page is rendering differently to those in production - can we change it to match prod (which is our usual style)?

Review app
image

Production
image

File: app/views/content/teacher-training-advisers/_article.html.erb [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    <p>
+      They can help you:
+    </p>
+
+    <ul>
+      <li>act on any feedback</li>
+      <li>make your application stronger</li>
+    </ul>
+
+    <h3>If you're thinking about returning to teaching</h3>
```
Comment (gemmadallmandfe) [inline app/views/content/teacher-training-advisers/_article.html.erb:80]: @plisovin @philspray 

Apostrophe style needs changing

File: app/views/content/teacher-training-advisers/_ending.html.erb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+    <h3>If you're a veteran</h3>
```
Comment (gemmadallmandfe) [inline app/views/content/teacher-training-advisers/_ending.html.erb:1]: @plisovin @philspray Apostrophe style needs changing

**Review by gemmadallmandfe — APPROVED** [review]
Looks good to me - happy for this to go into prod

**Review by asatwal — APPROVED** [review]
👍 LGTM. Are there any plans to add feature / capybara tests ? I believe these should be added as soon as we can if time doesn't allow now.

**Comment by plisovin** [comment]
> 👍 LGTM. Are there any plans to add feature / capybara tests ? I believe these should be added as soon as we can if time doesn't allow now.

Thanks. The tests already there, for instance [[they](https://github.com/DFE-Digital/get-into-teaching-app/blob/master/spec/features/teacher_training_adviser/sign_up_spec.rb)](https://github.com/DFE-Digital/get-into-teaching-app/blob/master/spec/features/teacher_training_adviser/sign_up_spec.rb) just test it from another entry point

**Pull request merged**